src/snowflake/snowpark/dataframe.py

```diff
@@ -611,6 +612,20 @@ def __init__(
         self._statement_params = None
         self.is_cached: bool = is_cached  #: Whether the dataframe is cached.
 
+        # Whether all columns are VARIANT data type,
+        # which support querying nested fields via dot notations
+        # If SQL simplifier is enabled, we need to get logical plan from plan.from_
+        if isinstance(plan, (SelectStatement, MockSelectStatement)):
+            self._all_variant_cols = (
+                isinstance(plan.from_, SelectSnowflakePlan)
+                and isinstance(plan.from_.snowflake_plan.source_plan, ReadFileNode)
+                and plan.from_.snowflake_plan.source_plan.xml_reader_udtf is not None
+            )
```

Comment:
wouldn't we be missing some cases here where dataframe might have all variant cols but it is not created from a read file operation? If we only care about the cases where source is a `ReadFileNode`, should we rename the internal variable to say that?

Reply:
for now, we only care about the dataframe generated by reading a XML file (see https://docs.google.com/document/d/11J2bL6QTpDs0DnMJZAq5EUNUCdipB9kfzojdlX_sCuU/edit?tab=t.0#heading=h.gky25x9oveeg), so querying nested fields in variant data using dot notation can be supported. In the future, we might also want to support such dot notation when all columns are variant (e.g., we can run a describe query to get types, which is an extra cost), so I had this variable name